IM_calculation/Advanced_IM/Models/Burks_Baker_2013/run.py

```diff
@@ -0,0 +1,94 @@
+from IM_calculation.IM.intensity_measures import get_geom
+from IM_calculation.IM.Burks_Baker_2013_elastic_inelastic import Bilinear_Newmark_withTH
+from IM_calculation.IM import read_waveform, intensity_measures
+
+from qcore.timeseries import read_ascii
+from qcore import constants
+from qcore.im import order_im_cols_df
+
+model_dir = os.path.dirname(__file__)
+
+COMPONENT = ["090", "000"]
+
+STORIES = 10
+
+z_list = [0.05]  # damping ratio
+alpha_list = [0.05]  # strain hardening ratio
+dy_list = [0.1765, 0.4, 0.6]  # strain hardening ratio
+dt_list = [0.005]
+period = np.array(constants.DEFAULT_PSA_PERIODS)
+
+
+def main(comp_000, comp_090, output_dir):
+
+    os.makedirs(output_dir, exist_ok=True)
+
+    log_file = os.path.join(output_dir, "log")
+    logging.basicConfig(
+        format="%(asctime)s %(message)s", filename=log_file, level=logging.DEBUG
+    )
+
+    waveforms = {}
+    waveforms["000"], meta = read_ascii(comp_000, meta=True)
+    waveforms["090"] = read_ascii(comp_090)
+    DT = meta["dt"]
+    NT = meta["nt"]
+
+    results = {}
+
+    accelerations = np.array((waveforms["090"], waveforms["000"])).T
+    ordered_columns_dict = {}
+
+    for z in z_list:
+        for dt in dt_list:
+            for alpha in alpha_list:
+                for dy in dy_list:
+                    displacements = intensity_measures.get_SDI_nd(
+                        accelerations, period, NT, DT, z, alpha, dy, dt
+                    )
+                    sdi_values = np.max(np.abs(displacements), axis=1)
+                    im_names = []
+                    for t in period:
+                        im_name = f"SDI_{t}_dy{dy}"  # we only vary dy at the moment
+                        im_names.append(im_name)
+                        # for each period, we will have a list of im_names varying dy. to help ordering columns later
+                        if t not in ordered_columns_dict:
+                            ordered_columns_dict[t] = []
+                        ordered_columns_dict[t].append(im_name)
+
+                    for i, component in enumerate(COMPONENT):
+                        if component not in results:
+                            results[component] = {}
+                        for j, im_name in enumerate(im_names):
+                            results[component][im_name] = (
+                                sdi_values[j, i] * 100
+                            )  # Burks & Baker returns m, but output is stored in cm
+
+    ordered_columns = []
+    for t in period:
+        ordered_columns.extend(
+            ordered_columns_dict[t]
+        )  # order by period first, then by dy
+
+    im_csv_fname = os.path.join(output_dir, "Burks_Baker_2013.csv")
+    df = pd.DataFrame.from_dict(results, orient="index")
+    df.index.name = "component"
+    geom = pd.Series(get_geom(df.loc["000"], df.loc["090"]), name="geom")
+    df = df.append(geom)
+    df[ordered_columns].to_csv(im_csv_fname)
```

Comment:
I am really not sure what this means..?

Reply:
@jasonmotha  Welcome back and could you have a look at this PR again?

Reply:
@jasonmotha ?

Reply:
@jasonmotha ?

Reply:
Because each value in ordered_columns is f"SDI_{t}_dy{dy}", it doesn't allow for multiple dt, z or alpha values, which are currently taken from a list. We need to go back to Vahid to find out how he wants to handle multiple dt, z and alpha values. 
If he doesn't, then we either need to make dt, z and alpha single value, or put them in the im name anyway and make him a post processing script to strip them out.

Reply:
Good point - he is still not sure if he will need multiple values. The current structure gives him an option to try things out. Valid point on the column names.

Reply:
@jpa198  Added a change to show all variables in column names

Reply:
@jasonmotha is it acceptable now?

Reply:
@jpa198 / @sungeunbae Any reason for the explicit ordering (storing the order in a list) rather than just sorting the column names afterwards?

Reply:
@jasonmotha I don't remember the details - it was written quite a while ago, but I showed @jpa198 when I had a great deal of trouble with the sorting function to keep the sensible numeric/ alphabetical order.

Reply:
I think you showed me that too; wasn't the end of that issue that the df was getting written twice? Once was sorted, and then the second time wasn't and kept overwriting the sorted one?

Reply:
No not that one. That one was a case where no change of order seen before and after. This one was where the order was mixed up with the available sorting method when alphabet and numeric values are all there. You know 1, 10, 2..when you want 1,2,10. I tried natural sort and other ways and nothing worked. Enlighten me.

Reply:
I haven't seen the exact csv so not sure if this solution works for you as well:
https://github.com/ucgmsim/IM_calculation/blob/36f25f00157fb3382570c36a6682a2002a3d6bc8/IM_calculation/Advanced_IM/advanced_IM_factory.py#L156

Reply:
I tried that too and didnt work. cant remember the details. It's been 20 days and I never thought this was going to be a hot talking point. One thing I remember, a simple sort worked better for these column names than this natural sort, and it almost worked, except this example. Uploading csv shortly

Reply:
[Burks_Baker_2013.csv](https://github.com/ucgmsim/IM_calculation/files/6919907/Burks_Baker_2013.csv)
<img width="811" alt="Screen Shot 2021-08-03 at 9 24 03 AM" src="https://user-images.githubusercontent.com/466989/127930983-1ea29d92-70f1-458b-af78-005f980bff6c.png">

Reply:
My main concern was to reduce complexitity by leveraging built-ins rather than having to keep track of it manually, if it's not working easily. Then happy as is.